reaper/cluster.go

```diff
@@ -0,0 +1,243 @@
+package reaper
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"net/http"
+	"net/url"
+	"runtime"
+	"sync"
+	"time"
+)
+
+type Cluster struct {
+	Name           string
+	JmxUsername    string
+	JmxPasswordSet bool
+	Seeds          []string
+	NodeState      NodeState
+}
+
+type NodeState struct {
+	GossipStates []GossipState
+}
+
+type GossipState struct {
+	SourceNode    string
+	EndpointNames []string
+	TotalLoad     float64
+	DataCenters   map[string]DataCenterState
+}
+
+type DataCenterState struct {
+	Name  string
+	Racks map[string]RackState
+}
+
+type RackState struct {
+	Name      string
+	Endpoints []EndpointState
+}
+
+type EndpointState struct {
+	Endpoint       string
+	DataCenter     string
+	Rack           string
+	HostId         string
+	Status         string
+	Severity       float64
+	ReleaseVersion string
+	Tokens         string
+	Load           float64
+}
+
+type GetClusterResult struct {
+	Cluster *Cluster
+	Error   error
+}
+
+type RepairSchedule struct {
+	Id                string    `json:"id"`
+	Owner             string    `json:"owner,omitempty"`
+	State             string    `json:"state,omitempty"`
+	Intensity         float64   `json:"intensity,omitempty"`
+	ClusterName       string    `json:"cluster_name,omitempty"`
+	KeyspaceName      string    `json:"keyspace_name,omitempty"`
+	RepairParallism   string    `json:"repair_parallelism,omitempty"`
```

Comment:
Is the Repair prefix needed here, or can we generalize this as just Parallelism?

Reply:
It could be confusing to remove `Repair` as parallelism could mean different things. I personally like the idea of sticking to the JSON name as much as possible.
BTW, `RepairParallism` has a typo which I think is inherited from Reaper itself :) Maybe it could be good to rename it here to `RepairParallelism`?

Reply:
Whoops sorry, done! Thanks for spotting this.

Reply:
Actually this was legacy code. But fixed anyways.